packages/cli/test/integration/credentials.ee.test.ts

```diff
@@ -406,14 +410,49 @@ describe('PUT /credentials/:id/share', () => {
 		expect(response.statusCode).toBe(403);
 	});
 
-	test('should respond 403 for non-owned credentials', async () => {
+	test('should respond 403 for non-owned credentials for shared members', async () => {
 		const savedCredential = await saveCredential(randomCredentialPayload(), { user: member });
 
-		const response = await authOwnerAgent
+		await shareCredentialWithUsers(savedCredential, [anotherMember]);
+
+		const response = await authAnotherMemberAgent
 			.put(`/credentials/${savedCredential.id}/share`)
-			.send({ shareWithIds: [member.id] });
+			.send({ shareWithIds: [owner.id] });
 
 		expect(response.statusCode).toBe(403);
+		const sharedCredentials = await Container.get(SharedCredentialsRepository).find({
+			where: { credentialsId: savedCredential.id },
+		});
+		expect(sharedCredentials).toHaveLength(2);
+	});
+
+	test('should respond 403 for non-owned credentials for non-shared members', async () => {
+		const savedCredential = await saveCredential(randomCredentialPayload(), { user: member });
+
+		const response = await authAnotherMemberAgent
+			.put(`/credentials/${savedCredential.id}/share`)
+			.send({ shareWithIds: [anotherMember.id] });
```

Comment:
I expect we want to test a non-sharee sharing a non-owned cred with someone other than themselves? Same in other suite.